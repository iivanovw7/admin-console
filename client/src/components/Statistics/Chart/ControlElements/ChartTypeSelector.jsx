import {
  FormControl,
  FormHelperText,
  Input,
  InputLabel,
  MenuItem,
  Select
} from '@material-ui/core';
import PropTypes from 'prop-types';
import React from 'react';

export const ChartTypeSelector = props => {

  const { classes, option, options, handleSelect } = props;

  return (
    <FormControl className={classes.formControl}>
      <InputLabel>Chart</InputLabel>
      <Select
        value={option.id}
        onChange={handleSelect}
        disabled={false}
        input={<Input name="style" id="charts-style"/>}
      >
        {options.map(newOption => (
          <MenuItem key={newOption.id} value={newOption.id}>
            {newOption.type}
          </MenuItem>
        ))}
      </Select>
      <FormHelperText>Select styles</FormHelperText>
    </FormControl>

  );
};

ChartTypeSelector.propTypes = {
  handleSelect: PropTypes.func.isRequired,
  options: PropTypes.array.isRequired
};

