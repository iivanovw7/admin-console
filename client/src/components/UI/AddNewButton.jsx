import React from 'react';
import PropTypes from 'prop-types';
import Fab from '@material-ui/core/Fab';
import AddIcon from '@material-ui/icons/Add';
import Button from '@material-ui/core/Button';
import withWidth from '@material-ui/core/withWidth';

const AddNewButton = props => {

  const { classes, width, history, element } = props;

  function handleClick() {
    return history.push(history.push(`/${element}/new`));
  }

  const mobileButton = () => {
    return (
      <Button size="small" className={classes.margin}
              onClick={() => {
                handleClick();
              }}>
        NEW
      </Button>
    );
  };

  const desktopButton = () => {
    return (
      <Fab color="primary" size="small" aria-label="Add" className={classes.fab}
           onClick={() => {
             handleClick();
           }}>
        <AddIcon/>
      </Fab>
    );
  };

  const components = {
    xs: mobileButton(),
    sm: mobileButton()
  };

  const Component = components[width] || desktopButton();

  return (
    <div className={classes.formRoot}>
      {Component}
    </div>
  );

};

AddNewButton.propTypes = {
  width: PropTypes.string.isRequired
};

export default withWidth()(AddNewButton);

