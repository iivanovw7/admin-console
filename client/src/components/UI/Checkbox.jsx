import Checkbox from '@material-ui/core/Checkbox';
import PropTypes from 'prop-types';
import React from 'react';

export const ActionCheckBox = props => {

  const { status, classes, handleClick } = props;

  return (
    <Checkbox
      className={classes.actionCheckBox}
      checked={status}
      size={'small'}
      color={'primary'}
      onChange={handleClick}
      disabled={false}
    />
  );

};

ActionCheckBox.propTypes = {
  classes: PropTypes.object,
  handleClick: PropTypes.func,
  status: PropTypes.bool
};
