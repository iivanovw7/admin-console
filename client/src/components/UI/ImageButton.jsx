import { Button, IconButton, ListItem } from '@material-ui/core';
import PropTypes from 'prop-types';
import React from 'react';

export const ImageButton = props => {

  const { classes, image, alt, handleClick, title } = props;

  return (
    <ListItem style={{ padding: 0, marginTop: 3, marginBottom: 3 }}>
      <IconButton className={classes.desktop} color="default" aria-label={alt} onClick={handleClick}
                  size={'small'}>
        <img style={{ width: '24px' }} src={image} alt={alt}/>
      </IconButton>
      <Button
        style={{ fontSize: '0.5rem' }}
        className={classes.mobile} size="small"
        onClick={handleClick}
      >
        {title}
      </Button>
    </ListItem>
  );
};

ImageButton.propTypes = {
  handleClick: PropTypes.func.isRequired,
  image: PropTypes.any.isRequired
};

