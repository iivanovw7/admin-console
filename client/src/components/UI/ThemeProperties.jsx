import blue from '@material-ui/core/colors/blue';
import { fade } from '@material-ui/core/styles/colorManipulator';
import { createMuiTheme } from '@material-ui/core/styles';

const drawerWidth = 240;

//Exporting material-ui main themeProperties configuration
export const themeProperties = createMuiTheme({
  typography: {
    useNextVariants: true
  },
  mixins: {
    toolbar: {
      textAlign: 'center',
      minHeight: '64px',
      justifyContent: 'space-between'
    }
  },
  palette: {
    primary: blue,
    secondary: {
      main: '#ffffff'
    }
  },
  status: {
    danger: 'orange'
  }
});

//Exporting material-ui main styles
export const NavigationStyles = theme => ({
  active: {
    backgroundColor: 'rgba(0, 0, 0, 0.10)'
  },
  root: {
    display: 'flex'
  },
  drawer: {
    [theme.breakpoints.up('sm')]: {
      width: drawerWidth,
      flexShrink: 0
    }
  },
  appBar: {
    marginLeft: drawerWidth,
    [theme.breakpoints.up('sm')]: {
      width: `calc(100% - ${drawerWidth}px)`
    }
  },
  menuButton: {
    marginRight: 20,
    [theme.breakpoints.up('sm')]: {
      display: 'none'
    }
  },
  toolbar: theme.mixins.toolbar,
  toolbarUserName: {
    [theme.breakpoints.down('xs')]: {
      display: 'none'
    }
  },
  drawerPaper: {
    width: drawerWidth
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3
  },
  search: {
    position: 'relative',
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.15),
    '&:hover': {
      backgroundColor: fade(theme.palette.common.white, 0.25)
    },
    marginRight: theme.spacing.unit * 2,
    marginLeft: 0,
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      marginLeft: theme.spacing.unit * 3,
      width: 'auto'
    }
  },
  searchIcon: {
    width: theme.spacing.unit * 9,
    height: '100%',
    position: 'absolute',
    pointerEvents: 'none',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center'
  },
  inputRoot: {
    color: 'inherit',
    width: '100%'
  },
  inputInput: {
    paddingTop: theme.spacing.unit,
    paddingRight: theme.spacing.unit,
    paddingBottom: theme.spacing.unit,
    paddingLeft: theme.spacing.unit * 10,
    transition: theme.transitions.create('width'),
    width: '100%',
    [theme.breakpoints.up('md')]: {
      width: 200
    }
  }
});

export const LoginFormStyles = theme => ({

  [theme.breakpoints.down('sm')]: {
    loginFormContainer: {
      boxShadow: 'none',
      height: '100%',
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center'
    }
  },
  loginWrapper: {
    maxWidth: '500px',
    margin: '0 auto',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    height: '100%'
  },
  loginFormContainer: {
    padding: theme.spacing.unit,
    maxWidth: '500px'
  },
  formPaper: {
    margin: theme.spacing.unit * 2
  },
  paddingTop: theme.spacing.unit * 2,
  paddingBottom: theme.spacing.unit * 2

});

//Exporting material-ui main styles
export const Wrapper = theme => ({
  root: {
    display: 'flex'
  },
  toolbar: theme.mixins.toolbar,
  contentSingle: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3,
    marginTop: '64px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    height: '100%',
  },
  contentList: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3,
    marginTop: '64px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    [theme.breakpoints.up('sm')]: {
      height: '90vh'
    },
    [theme.breakpoints.down('sm')]: {
      height: '100%',
    }
  },
  formRoot: {
    display: 'flex',
    flexWrap: 'wrap',

    buttonWrapper: {
      display: 'flex',
      alignItems: 'center'
    }
  },
  fromLimitSelector: {
    display: 'flex',
    flexWrap: 'wrap',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  formControl: {
    margin: theme.spacing.unit,
    minWidth: 120,
    maxWidth: 300
  },
  controlsContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingRight: 10,
    paddingLeft: 10
  },
  selectorsContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    paddingRight: 10,
    paddingLeft: 10
  }
});

export const Container = theme => ({
  root: {
    width: '100%',
    marginTop: theme.spacing.unit * 3,
    overflowX: 'auto',
    display: 'flex',
    justifyContent: 'center',
    overflow: 'hidden'
  },
  tables: {
    minWidth: 200,
    width: '100%'
  },
  tableCell: {
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5
  },
  rowClass: {
    '&:hover': {
      backgroundColor: fade(theme.palette.common.black, 0.1),
      cursor: 'pointer'
    },
    [theme.breakpoints.up('sm')]: {
      height: '48px',
    },
    [theme.breakpoints.down('sm')]: {
      height: '100%',
    },
  },
  nameCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,

    [theme.breakpoints.up('sm')]: {
      width: '30%',
    },

    [theme.breakpoints.down('sm')]: {
      width: '60%',
    }

  },
  addressCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,
    width: '100%',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  employeesCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,
    width: '30%',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  branchPaper: {
    width: '100%',
    maxWidth: 600,
    margin: theme.spacing.unit * 2
  }
});

export const Notification = theme => ({
  close: {
    padding: theme.spacing.unit / 2
  }
});

export const Buttons = theme => ({
  formRoot: {
    display: 'flex',
    flexWrap: 'wrap'
  },
  [theme.breakpoints.down('sm')]: {
    mobile: {
      display: 'flex'
    },
    desktop: {
      display: 'none'
    }
  },
  [theme.breakpoints.up('sm')]: {
    mobile: {
      display: 'none'
    },
    desktop: {
      display: 'flex'
    }
  }
});

