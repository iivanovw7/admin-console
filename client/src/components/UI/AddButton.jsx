import { withStyles } from '@material-ui/core';
import React from 'react';
import PropTypes from 'prop-types';
import Fab from '@material-ui/core/Fab';
import AddIcon from '@material-ui/icons/Add';
import Button from '@material-ui/core/Button';
import { Buttons } from './ThemeProperties';

const AddButton = props => {

  const { classes, history, element } = props;

  function handleClick() {
    return history.push(history.push(`/${element}/new`));
  }

  return (
    <div className={classes.formRoot}>
      <Button size="small" className={classes.mobile}
              onClick={() => {
                handleClick();
              }}>
        <strong>Create</strong>
      </Button>
      <Fab color="primary" size="small" aria-label="Add" className={classes.desktop}
           onClick={() => {
             handleClick();
           }}>
        <AddIcon/>
      </Fab>
    </div>
  );

};

AddButton.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(Buttons, { withTheme: true })(AddButton);