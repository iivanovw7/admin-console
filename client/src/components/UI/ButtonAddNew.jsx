import { Button, Fab } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import AddIcon from '@material-ui/icons/Add';
import PropTypes from 'prop-types';
import React from 'react';
import { Buttons } from './ThemeProperties';

const ButtonAddNew = props => {

  const { classes, history, element } = props;

  function handleClick() {
    return history.push(history.push(`/${element}/new`));
  }

  return (
    <div className={classes.formRoot}>
      <Button
        size="small"
        className={classes.mobile}
        onClick={() => {
          handleClick();
        }}
      >
        <strong>Create</strong>
      </Button>
      <Fab
        color="primary"
        size="small"
        aria-label="Add"
        className={classes.desktop}
        onClick={() => {
          handleClick();
        }}>
        <AddIcon/>
      </Fab>
    </div>
  );

};

ButtonAddNew.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(Buttons, { withTheme: true })(ButtonAddNew);