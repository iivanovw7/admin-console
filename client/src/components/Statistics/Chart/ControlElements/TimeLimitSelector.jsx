import PropTypes from 'prop-types';
import React from 'react';
import { NumbersSelector } from '../../../UI/NumbersSelector';

export const TimeLimitSelector = props => {
  const { classes, dataType, limit, limits, handleLimit } = props;
  const hideSelector = ['Permissions', 'Groups'].indexOf(dataType) > -1;

  return (
    <NumbersSelector
      classes={classes}
      title={'Months'}
      option={hideSelector ? 'Total' : limit}
      options={hideSelector ? ['Total'] : limits}
      disabled={hideSelector}
      handleSelect={handleLimit}
    />
  );
};

TimeLimitSelector.propTypes = {
  handleLimit: PropTypes.func.isRequired,
  limits: PropTypes.array.isRequired
};
