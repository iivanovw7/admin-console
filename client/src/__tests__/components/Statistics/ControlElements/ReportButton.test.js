import { withStyles, IconButton } from '@material-ui/core';
import { createMount, createShallow } from '@material-ui/core/test-utils';
import { createMemoryHistory } from 'history';
import jest from 'jest-mock';
import React from 'react';
import csvIcon from '../../../../icons/csv-file-format-symbol.svg';
import { TopInfoBarButton } from '../../../../components/Statistics/Chart/ControlElements/ReportButton';
import { Container } from '../../../../components/UI/ThemeProperties';

const setup = () => {
  let wrapper;
  let shallowWithDive;
  let mount;
  let shallow;
  const mockOnClick = jest.fn();
  const history = createMemoryHistory('/dashboard');
  const Composition = withStyles(Container, { withTheme: true })(TopInfoBarButton);

  const props = {
    history: history,
    image: csvIcon,
    alt: 'test-alt',
    handleClick: mockOnClick,
    title: 'TEST'
  };

  beforeEach(() => {
    shallow = createShallow();
    shallowWithDive = createShallow({ dive: true });
    mount = createMount();
  });

  afterEach(() => {
    mount.cleanUp();
  });

  describe('Should render Button with styles and options', () => {

    it('Should render component', () => {
      wrapper = mount(<Composition {...props} />);
      expect(wrapper.find('li')).not.toBe(undefined);
      expect(wrapper.find('li').html()).not.toBe(null);
      expect(wrapper.find('li').length).toEqual(1);
      expect(wrapper.find('button').length).toEqual(2);
    });

    it('Should render desktop button', () => {
      wrapper = mount(<Composition {...props} />);
      expect(wrapper.find(IconButton)).not.toBe(undefined);
      expect(wrapper.find(IconButton)).not.toBe(null);
    });

    it('Should render mobile button', () => {
      wrapper = mount(<Composition {...props} />);
      expect(wrapper.find('button').last()).not.toBe(undefined);
      expect(wrapper.find('button').last().html()).not.toBe(null);
      expect(wrapper.find('button').last().text()).toBe('TEST');

    });

    it('Click handling', () => {
      wrapper = mount(<Composition {...props} />);
      wrapper.find(IconButton).simulate('click');
      expect(mockOnClick).toHaveBeenCalled();
    });

    it('Should match snapshot', () => {
      wrapper = mount(<Composition {...props} />);
      expect(wrapper.html()).toMatchSnapshot();
    });


  });

};

setup();



