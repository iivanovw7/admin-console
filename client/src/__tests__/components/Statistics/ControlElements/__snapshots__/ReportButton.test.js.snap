// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Button with styles and options Should match snapshot 1`] = `"<li class=\\"MuiListItem-root-347 MuiListItem-default-350 MuiListItem-gutters-355\\" style=\\"padding: 0px; margin-top: 3px; margin-bottom: 3px;\\"><button class=\\"MuiButtonBase-root-365 MuiIconButton-root-359 TopInfoBarButton-desktop-324\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"test-alt\\"><span class=\\"MuiIconButton-label-364\\"><img style=\\"width: 24px;\\" src=\\"csv-file-format-symbol.svg\\" alt=\\"test-alt\\"></span><span class=\\"MuiTouchRipple-root-394\\"></span></button><button class=\\"MuiButtonBase-root-365 MuiButton-root-368 MuiButton-text-370 MuiButton-flat-373 MuiButton-sizeSmall-391 TopInfoBarButton-mobile-322\\" tabindex=\\"0\\" type=\\"button\\" style=\\"font-size: 0.5rem;\\"><span class=\\"MuiButton-label-369\\">TEST</span><span class=\\"MuiTouchRipple-root-394\\"></span></button></li>"`;
